### PR TITLE
Normalize all qualified imports to post qualified (when using ImportQualifiedPost)

### PR DESCRIPTION
**Pull request by beefyhalo**
If we are using post qualified imports in a file, turn the regular qualified ones into post qualified ones.

### CODE REVIEW

File: src/SimSpace/SimFormat.hs [hunk 1]
```diff
@@ -483,12 +484,26 @@ reformat regroup programLines = do
...
           <> (fmap . fmap) renderImportStmts rechunked
           <> leftovers
       where
+        -- If we are using post qualified imports, turn the regular qualified ones into post qualified ones
+        requalify :: [Either BlankLine SortedImportStmts] -> [Either BlankLine SortedImportStmts]
+        requalify ci
+          | any hasPostQualified (rights ci) =
+            fmap (SortedImportStmts . Map.mapKeys requalifyImportStmt . unSortedImportStmts) <$> ci
+          | otherwise = ci
+          where
```
Comment (felixminom) [inline src/SimSpace/SimFormat.hs:499]: didn't know you can have nested wheres jeje

File: src/SimSpace/SimFormat.hs [hunk 2]
```diff
@@ -483,12 +484,26 @@ reformat regroup programLines = do
...
+            hasPostQualified :: SortedImportStmts -> Bool
+            hasPostQualified = any importStmtHeadPostQualified . Map.keys . unSortedImportStmts
+            requalifyImportStmt :: ImportStmtHead -> ImportStmtHead
+            requalifyImportStmt stmt
+              | importStmtHeadQualified stmt = stmt {importStmtHeadQualified = False, importStmtHeadPostQualified = True}
```
Comment (felixminom) [inline src/SimSpace/SimFormat.hs:504]: Is this enough to perform the post qualification? :nice:

**Review by felixminom — APPROVED** [review]
lgtm, was this tested locally?

File: src/SimSpace/SimFormat.hs (hunk 2 above)
Comment (beefyhalo) [reply src/SimSpace/SimFormat.hs:504]: The post qualify support is already in there ;) This bit just renormalizes non post-qualified ones to post-qualified if its enabled in the module. (So we don't end up with a mix of post and non-post ones in the same file)

**Pull request merged**